File: 36-valid-sudoku/36-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
            rows = defaultdict(set)
            cols = defaultdict(set)
            squares = defaultdict(set) #key (row/3, col/3)

            for i in range(len(board)):
                for j in range(len(board)):
                    if board[i][j] == '.':
                        continue
                    if (board[i][j] in rows[i] or
                        board[i][j] in cols[j] or
                        board[i][j] in squares[(i//3 , j//3)]):
                        return False

                    rows[i].add(board[i][j])
                    cols[j].add(board[i][j])
                    squares[(i//3 , j//3)].add(board[i][j])

            return True

File: 36-valid-sudoku/36-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        
	unordered_map<int , set<char>> rows;
	unordered_map<int , set<char>> cols;
	map<pair<int , int> , set<char>> squares;
		
	for(int i = 0 ; i < (int) board.size() ; i ++ ){
		for (int j = 0 ; j < (int)board.size() ; j++){
			if(board[i][j] == '.')continue;
			
			if(rows[i].find(board[i][j]) != rows[i].end() || 
			   cols[j].find(board[i][j]) != cols[j].end() ||
			   squares[{i/3,j/3}].find(board[i][j]) != squares[{i/3,j/3}].end())
				return false ;
			
			rows[i].insert(board[i][j]);
			cols[j].insert(board[i][j]);
			pair<int , int> p = {i/3 , j/3};
			squares[p].insert(board[i][j]);
			
		}
	}


	return true;

    }
};